executor-core/src/main/java/net/spals/appbuilder/executor/core/DefaultExecutorServiceFactory.java

```diff
@@ -105,24 +100,46 @@ public ScheduledExecutorService createScheduledThreadPool(
         return scheduledExecutorService;
     }
 
-    ExecutorService decorateExecutorService(final ExecutorService delegate) {
+    @Override
+    public Optional<Boolean> stop(final Key key) {
+        Objects.requireNonNull(key);
+        final Optional<ExecutorService> executoreService = Optional.ofNullable(executorServices.get(key));
+        if (executoreService.isPresent()) {
```

Comment:
I feel like I could be slicking with my options here to have no if statements but I couldn't figure it out.

Reply:
1. Have `stopExecutorService` return a boolean
2. `Optional.ofNullable(executorServices.get(key)).map(executorService -> stopExecutorService(key, executorService);`